fs/sync.go

```diff
@@ -21,6 +21,7 @@ type syncCopyMove struct {
 	noTraverse     bool              // if set don't trafevers the dst
 	deleteBefore   bool              // set if we must delete objects before copying
 	trackRenames   bool              // set if we should do server side renames
+	filesMu        sync.RWMutex      // Mutex covering the src and dst files maps
```

Comment:
You could define this as a local variable to renameFiles couldn't you?  Your go routines will close over it just fine.